### PR TITLE
fixed dockerfile lnd inside docker folder

### PR DESCRIPTION
**Pull request by guilhermehas**

### CODE REVIEW

**Comment by halseth** [comment]
What does this fix? Should have a description.

Also, might conflict with #2259

**Comment by wpaulino** [comment]
Looks like it can actually replace #2259. This PR will allow us to no longer assume the source code is present within the current directory.

**Comment by OluAgunloye** [comment]
Works, in conjunction w/ #2334 , on Ubuntu 16.04

**Comment by tuxx42** [comment]
i'd suggest the method proposed in my PR #2334 instead.
However, in #2334 one needs to build the containers from the root of `lnd` as such, but its better than cloning the git repo :
`host@~/lnd/$ sudo docker-compose -f docker/docker-compose.yml build lnd`

but i think this is more useful than cloning the repo, because this builds against whatever is currently checked out in the root of the `lnd` directory (e.g. a feature branch)

**Comment by guilhermehas** [comment]
After writing this command in the guide
`docker-compose run -d --name alice lnd_btc`
I got this error:
`ERROR: Service 'lnd_btc' failed to build: The command '/bin/sh -c apk add --no-cache     git     make &&  cd /go/src/github.com/lightningnetwork/lnd &&  make &&  make install' returned a non-zero code: 2 (edited) `

File: docker/lnd/Dockerfile [hunk 1]
```diff
@@ -1,18 +1,15 @@
-FROM golang:1.10-alpine as builder
-
-MAINTAINER Olaoluwa Osuntokun <lightning.engineering>
```
Comment (yancyribbens) [inline docker/lnd/Dockerfile]: Is there a reason you removed the Maintainer contact?

File: docker/lnd/Dockerfile [hunk 2]
```diff
@@ -1,18 +1,15 @@
...
-
-# Copy in the local repository to build from.
-COPY . /go/src/github.com/lightningnetwork/lnd
```
Comment (yancyribbens) [inline docker/lnd/Dockerfile]: Is there a reason to clone from the repo on L12 (which requires http connection) instead of using COPY?

**Comment by yancyribbens** [comment]
> Looks like it can actually replace #2259. This PR will allow us to no longer assume the source code is present within the current directory.

@wpaulino yes, but now to build you add a dependacy of needing http and more importantly GitHub.  I think this actually makes the build more brittle.

**Comment by yancyribbens** [comment]
@wpaulino yeah Github is used other places to pull files into the container.  Either options work, I just think it adds extra complexity since the state of github may be different than the state locally, and also github could go down which does happen from time to time.  Also, build time is slightly faster if not relying on an external service to pull int files that could otherwise be provided locally.  The net result is the same as a way to copy files into a docker container.

**Comment by Roasbeef** [comment]
Superseded by #2259

**Pull request closed without merging**